addons/project_timesheet_holidays/models/res_config_settings.py

```diff
@@ -1,11 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import fields, models,api
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     leave_timesheet_project_id = fields.Many2one(related='company_id.leave_timesheet_project_id', string="Internal Project", domain="[('company_id', '=', company_id)]", readonly=False)
     leave_timesheet_task_id = fields.Many2one(related='company_id.leave_timesheet_task_id', string="Time Off Task", domain="[('company_id', '=', company_id)]", readonly=False)
+
+    @api.onchange('leave_timesheet_project_id')
+    def _on_change_timesheet_project_id(self):
+        if self.leave_timesheet_project_id != self.leave_timesheet_task_id.project_id:
+            self.leave_timesheet_task_id = False
```

Comment:
oO one more ^^

Reply:
The `leave_timesheet_task_id` field is a related field so I cannot convert it to a compute field. Then, this model is a transient one, that is just used to set some settings.
In this case, I think that it is acceptable to use an onchange as it is dedicated to a UI behavior.